### DISCUSSION TITLE
Custom component properties for accessing in middleware

### DISCUSSION THREAD

**Discussion by Geometrically** [nuxt/nuxt#18533] [Questions] [answered]
Hello!

I am making an authentication middleware, which redirects users to the login page if they are not authenticated. Before in Nuxt 2, I could define a property on a component like this, and access it from my middleware (in this case to mark if a page requires authentication.)
```js
export default {
     auth: true
}
```

The method to do this no longer works. Is there an alternative in Nuxt 3 to mark metadata for a page which can be accessed via middleware?

**Comment by TheAlexLichter** [comment, marked as answer]
Yes! [`definePageMeta`](https://nuxt.com/docs/api/utils/define-page-meta/) is what you are looking for

**Reply by Geometrically** [reply]
How can I access values defined in definePageMeta from my middleware? ideally I don't want to type the entire middleware for every file which needs to be authenticated

**Reply by Geometrically** [reply]
Oh never mind, it's in the docs! https://nuxt.com/docs/guide/directory-structure/middleware#example-a-named-route-middleware
